modules/internal/ChapelIO.chpl

```diff
@@ -830,6 +830,13 @@ module ChapelIO {
     }
   }
 
+  override proc LocaleModel.writeThis(f) {
+      // Most classes will define it like this:
+      //      f <~> name;
+      // but here it is defined thus for backward compatibility.
+      f <~> new ioLiteral("LOCALE") <~> _node_id;
```

Comment:
Since this is indepedent of locale model, and _node_id might be considered a private field, I think it should get the id by calling chpl_id()

Reply:
Ah, that's what you meant on the other PR.  I can do that!

Reply:
Should I do more extensive testing for numa as a result, since the `chpl_id()` method could dynamically dispatch if the `writeThis` method is not overridden?